src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/LuceneDocumentConverterBase.cs

```diff
@@ -118,7 +121,7 @@ public IDisposable SetDocument(LazyStringValue key, LazyStringValue sourceDocume
             int numberOfFields = GetFields(new DefaultDocumentLuceneWrapper(Document), key, sourceDocumentId, document, indexContext);
             if (_fields.Count > 0)
             {
-                shouldSkip = _indexEmptyEntries == false && numberOfFields <= 1; // there is always a key field, but we want to filter-out empty documents
+                shouldSkip = _indexEmptyEntries == false && numberOfFields <= _numberOfBaseFields; // there is always a key field, but we want to filter-out empty documents
```

Comment:
Can you please update the comment to explain what are default fields depending on index type? So someone not familiar won't need to figure it out on their own

Reply:
Done